lib/debug/console.rb

```diff
@@ -1,6 +1,39 @@
 # frozen_string_literal: true
 module DEBUGGER__
   class Console
+    COMPLETION_PROC = -> given do
+      thread_client = SESSION.managed_thread_clients.first
+      binding = thread_client.current_binding
+      frame_self = thread_client.current_frame&.self
+
+      if frame_self.nil?
+        next DEBUGGER__.commands.keys.grep(/\A#{given}/)
+      end
+
+      candidates =
+        case given
+        when /\A@\w/
+          frame_self.instance_variables
+        when /\A[A-Z]/
+          if frame_self.is_a?(Module)
+            frame_self.constants
+          else
+            frame_self.class.constants
+          end
+        when /^::([A-Z][^:\.\(\)]*)$/
+          # Absolute Constant or class methods
```

Comment:
@ko1 this case's implementation is borrowed from `irb`. I've added acknowledgement in the commit message, but I'm not sure if I should add one in code comment too?